File: animation/src/main/java/me/wyne/wutils/animation/runnable/MessageEffect.java
package me.wyne.wutils.animation.runnable;

import me.wyne.wutils.animation.AnimationRunnable;
import net.kyori.adventure.audience.Audience;
import net.kyori.adventure.text.Component;
import org.jetbrains.annotations.NotNull;

/**
 * Sends {@code message} to {@code audience}.
 */
public record MessageEffect(@NotNull Audience audience, @NotNull Component message) implements AnimationRunnable {

    @Override
    public void run() {
        audience.sendMessage(message);
    }

}

File: animation/src/main/kotlin/me/wyne/wutils/animation/runnable/MessageEffect.kt
package me.wyne.wutils.animation.runnable

import me.wyne.wutils.animation.AnimationRunnable
import net.kyori.adventure.audience.Audience
import net.kyori.adventure.text.Component

class MessageEffect(
    private val audience: Audience,
    private val message: Component
) : AnimationRunnable {

    override fun run() {
        audience.sendMessage(message)
    }

}